app/controllers/core_data_connector/public/v1/public_controller.rb

```diff
@@ -11,7 +11,7 @@ def base_query
           return item_class.none unless params[:project_ids].present?
 
           if nested_resource?
-            item_class.where(build_base_sql)
+            item_class.joins(build_base_sql(item_class.table_name))
```

Comment:
`item_class` is a protected method, so it should be accessible from `build_base_sql`, so probably no need to pass as an argument.